sqlglot/generator.py

```diff
@@ -1770,6 +1770,11 @@ def concat_sql(self, expression: exp.Concat) -> str:
             return self.sql(expression.expressions[0])
         return self.function_fallback_sql(expression)
 
+    def safeconcat_sql(self, expression: exp.SafeConcat) -> str:
+        if len(expression.expressions) == 1:
```

Comment:
why would len ever be 1 ? what’s the point of this check

Reply:
I changed this to delegate to `concat_sql`; the point of this check is to reduce something like `CONCAT('a')` to `'a'` (the former is valid MySQL, for example).

Reply:
why bother? that's a job for the optimizer

Reply:
it helps transpile `concat('a')` e.g. to presto/trino correctly, because it requires >1 arguments. There might be other dialects that have these semantics too, so it doesn't hurt to keep this imo.

```
trino> select concat('a');
Query 20230607_113101_00000_4ki82 failed: There must be two or more concatenation arguments
```